### PR TITLE
Added new TC DeleteBackupOfUserNonSharedRBAC

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Added new TC DeleteBackupOfUserNonSharedRBAC and Required Helper Functions
DeleteBackupOfUserNonSharedRBAC - https://portworx.testrail.net/index.php?/cases/view/87561

**Which issue(s) this PR fixes** (optional)
Closes #PA-1398

**Special notes for your reviewer**:
This PR doesn't contains 2.5.2 api
logs:
s3:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2406/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/323498

NFS:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/NFS/job/PX-Cloud-Vanilla-NFS/128/

https://aetos.pwx.purestorage.com/resultSet/testSetID/327353

### CODE REVIEW

**Review by snigdha-px — APPROVED** [review]
LGTM
Can we have a full run after all the testcases are done for 2.6.0 as I see  in the logs attached that one test case has failed for both s3 and NFS

File: tests/backup/backup_orphaned_delete_test.go [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+			log.InfoD("Create Users with Different types of roles")
+			roles := [2]backup.PxBackupRole{backup.ApplicationOwner, backup.InfrastructureOwner}
+			for i := 1; i <= numOfUsers/2; i++ {
+				for _, role := range roles {
+					userName := fmt.Sprintf("testauser-%v-%s", role, RandomString(4))
+					firstName := fmt.Sprintf("FirstName-%v-%s", role, RandomString(4))
+					lastName := fmt.Sprintf("LastName-%v-%s", role, RandomString(4))
+					email := fmt.Sprintf("testuser-%v-%s@cnbu.com", role, RandomString(4))
+					err := backup.AddUser(userName, firstName, lastName, email, commonPassword)
+					log.FailOnError(err, "Failed to create user - %s", userName)
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:126]: How about?

userName := createUsers(1)[0] just to be consistent?

File: tests/backup/backup_orphaned_delete_test.go [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+					}
+					userNames = append(userNames, userName)
+					userUID, err := backup.FetchIDOfUser(userName)
+					log.FailOnError(err, "Failed to fetch uid for - %s", userName)
+					userIdMap[userName] = userUID
+				}
+			}
+		})
+
+		Step(fmt.Sprintf("Adding cloud account for backup location from px-admin "), func() {
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:142]: nit "and" backup location

File: tests/backup/backup_orphaned_delete_test.go [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+					infraUserCredName = fmt.Sprintf("%v-cred-%v", provider, RandomString(5))
+					err = CreateCloudCredential(provider, infraUserCredName, infraUserCloudCredUID, orgID, ctx)
+					log.FailOnError(err, "Failed to create cloud account for backup location from px-admin user  - %s", err)
+					userCredNameMap[infraAdminUserName] = infraUserCredName
+					userCloudCredUIDMap[infraAdminUserName] = infraUserCloudCredUID
+				}
+			})
+		}
+
+		Step(fmt.Sprintf("Create backup location for non-admin user using the cloud cred "), func() {
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:180]: I think log.InfoD is missing.

**Comment by ak-px** [comment]
Thanks for the reviews. These changes are now included in PR https://github.com/portworx/torpedo/pull/1669.

**Pull request closed without merging**